proto/spacemesh/api.proto

```diff
@@ -1,166 +1,70 @@
 
-message AccountEvent {
-    enum Event {
-        UNDEFINED = 0;
-        BALANCE_CHANGE = 1;
-        NONCE_CHANGE = 2;
-    }
-    uint64 value = 1;
-}
+    // Returns current layer number
+    rpc MeshCurrentLayer(google.protobuf.Empty) returns (SimpleInt);
 
-message NodeStatus {
-    uint64 peers = 1;
-    uint64 min_peers = 2;
-    uint64 max_peers = 3;
-    bool synced = 4;
-    uint64 synced_layer = 5;
-    uint64 current_layer = 6;
-    uint64 verified_layer = 7;
-}
+    // Returns current epoch number
+    rpc MeshCurrentEpoch(google.protobuf.Empty) returns (SimpleInt);
 
-/*
-   Represents a critical node error.
-   TODO: add an enum with clearly-defined error types.
- */
-message NodeError {
-    string error = 1;
-}
+    // Returns network ID
+    rpc MeshNetId(google.protobuf.Empty) returns (SimpleInt);
 
-message SyncStatus {
-  enum SyncStatus {
-      UNDEFINED = 0;
-      NEW_LAYER_VERIFIED = 1;
-      NEW_TOP_LAYER = 2;
-      SYNCED = 3;
-      SYNCING = 4;
-      NOT_SYNCED = 5;
-  }
-}
+    // Returns number of layers per epoch (a network constant)
+    rpc MeshEpochNumLayers(google.protobuf.Empty) returns (SimpleInt);
 
-service SpacemeshService {
-    // SCALAR
+    ////////// Node streaming data facade
 
-    // Test facade
+    // Sync status events
+    rpc NodeSyncStatuses(google.protobuf.Empty) returns (stream NodeSyncStatus);
 
-    // A simple test endpoint
-    rpc Echo(SimpleMessage) returns (SimpleMessage);
+    // Node error events
+    rpc NodeErrors(google.protobuf.Empty) returns (stream NodeError);
 
-    // Node facade
+    //////////////// Mesh streaming data facade
 
-    // Returns the version of the node software as a semver string
-    rpc NodeVersion(google.protobuf.Empty) returns (SimpleMessage);
-    // Returns node status object
-    rpc NodeStatus(google.protobuf.Empty) returns (NodeStatus);
-    // Tell the node to start sync
-    rpc NodeSyncStart(google.protobuf.Empty) returns (google.protobuf.Empty);
+    // Receipts - emitted after tx was processed by STF (or rejected before STF)
+    rpc MeshTransactionReceipts(google.protobuf.Empty) returns (stream TransactionReceipt);
 
-    // Mesh facade
+    // App Events - emitted by app methods impl code trigged by an app transaction
+    rpc MeshAppEvents(google.protobuf.Empty) returns (stream AppEvent);
 
-    // Returns mesh genesis time (as UTC string)
-    rpc MeshGenesisTime(google.protobuf.Empty) returns (SimpleMessage);
-    // Returns current layer
-    rpc MeshCurrentLayer(google.protobuf.Empty) returns (SimpleInt);
-    // Returns current epoch
-    rpc MeshCurrentEpoch(google.protobuf.Empty) returns (SimpleInt);
-    // Returns network ID
-    rpc MeshNetId(google.protobuf.Empty) returns (SimpleInt);
-    // Returns number of layers per epoch
-    rpc MeshEpochNumLayers(google.protobuf.Empty) returns (SimpleInt);
+    // Accounts
+    rpc MeshAccounts(google.protobuf.Empty) returns (stream Account);
```

Comment:
If we replace `AccountEvent` with `Account`, it's no longer possible to tell directly from the stream what changed. It becomes contextual/stateful, i.e., you have to compare with the previous state to see what changed. Maybe this is okay, but we should pick one scheme and stick with it. Might be worth discussing with the explorer team. I'll open an issue.

Reply:
See my above comment - I think we can do without any Events per-se and it is a simpler design which is satisfactory. I don't think there's a way to get around clients having to look for state changes in data objects in stream - for example, transactions, layers, blocks can all change and be sent multiple times... It is actually good the node api is very low-level and doesn't indicate state changes just RAW data and the responsibility to understand state changes moves up to the data workers in client backends... So this is the scheme I think we should stick to and mock for now. We can always add data items state events methods later. This will also greatly simplify logic in the node as it doesn't need to know when a state that requires to raise an event occurs and deal with it...  This I think also the main way we handle reogrs elegantly. In a reorg, the layers, blocks and txs are just sent again over the standard entities streams and the client updates its date based on the current fields in these entities....